crates/oxc_semantic/src/label.rs

```diff
@@ -54,9 +57,18 @@ impl<'a> LabelBuilder<'a> {
         }
 
         self.depth += 1;
-
+        let mut parent_id = None;
+        for labels_in in self.labels.iter() {
+            for labels in labels_in.iter().rev() {
+                if labels.depth + 1 == self.depth {
+                    parent_id = Some(labels.id);
+                    break;
+                }
+            }
+        }
```

Comment:
Do you feel like the algorithm is inefficient overall? Although labels are rare to have any any impact on the overall performance. But we have so much code for handling labels, which is pretty weird 🤔

Reply:
> Do you feel like the algorithm is inefficient overall? Although labels are rare to have any any impact on the overall performance. But we have so much code for handling labels, which is pretty weird 🤔

Yes, I have thought about it before. Perhaps we can only keep the parent before increasing the depth to avoid traversing it every time.
I was thinking the same as you before, the possibility of labels being too complex is unlikely. I was afraid that if the solution of PR was wrong, there was no need to write it. Loop traversal seems easier to write.🤣